### PR TITLE
Dfw threshold

### PR DESCRIPTION
**Pull request by pandom**
DFW Threshold command

- Set-NsxFirewallThreshold
   - Added integer maximums per internal sourcecode.
- Get-NsxFirewallThreshold
- Pester tests - Context added to DFW Tests
- Added to PSD manifest

MISC:

Updated help texts of some commands.

Output of Pester.

`SNIP

Context DFW Global options

    [+] Can adjust default DFW event threshold 216ms

SNIP`

### CODE REVIEW

File: PowerNSX.psm1 [hunk 1]
```diff
@@ -22338,7 +22376,137 @@ function Get-NsxFirewallSavedConfiguration {
...
+                    "CPU" = $threshold.eventthresholds.cpu.percentvalue;
+                    "Memory" = $threshold.eventthresholds.memory.percentvalue;
+                    "ConnectionsPerSecond" = $threshold.eventThresholds.connectionsPerSecond.value
+                }
+    
+    }
+
+        
+       
+
```
Comment (nmbradford) [inline PowerNSX.psm1]: Clean up white space.  remove blank lines - retain single ones that separate a logical block or leading a comment.  Remove any multiple blank lines.

File: PowerNSX.psm1 [hunk 2]
```diff
@@ -22338,7 +22376,137 @@ function Get-NsxFirewallSavedConfiguration {
...
+
+    #>
+
+    param (
+
+        [Parameter (Mandatory=$true)]
+            [ValidateRange(1,100)]
+            [int]$Memory,
+        [Parameter (Mandatory=$true)]
```
Comment (nmbradford) [inline PowerNSX.psm1]: Are cpu, mem and cps mandatory?  check if API accepts single, if so - make optional.  Must test with PSBoundParameters.ContainsKey("cpu") et all before adding to xml later in cmdlet...

File: PowerNSX.psm1 [hunk 3]
```diff
@@ -22338,7 +22376,137 @@ function Get-NsxFirewallSavedConfiguration {
...
+    begin {
+
+         #Create the XMLRoot
+            [System.XML.XMLDocument]$xmlDoc = New-Object System.XML.XMLDocument
+            [System.XML.XMLElement]$xmlRoot = $XMLDoc.CreateElement("eventThresholds")
+            $xmlDoc.appendChild($xmlRoot) | out-null
+
+            #Create an Element and append it to the root
+
+            [System.XML.XMLElement]$xmlcpu = $xmlDoc.CreateElement("cpu")
```
Comment (nmbradford) [inline PowerNSX.psm1]: if param is optional - make sure you test for existence here...

**Comment by pandom** [comment]
Updated. 

All properties are optional and preserves existing settings if they aren't referenced.

Tests created to support new criteria such as CPU only, Memory Only, Connections Only.

```
Context DFW Global options
    [+] Can adjust all DFW event thresholds 200ms
    [+] Can adjust Memory DFW event threshold 186ms
    [+] Can adjust CPU DFW event threshold 187ms
    [+] Can adjust ConnectionsPerSecond DFW event thresholds 174ms
    [?] Can adjust Global Containers 3ms
    [?] Can adjust TCP Optimisation 3ms
```

**Pull request closed without merging**